custom/enikshay/integrations/nikshay/repeaters.py

```diff
@@ -34,17 +37,60 @@ def allowed_to_forward(self, episode_case):
         # When case property episode.episode_pending_registration transitions from 'yes' to 'no',
         # and (episode.nikshay_registered != 'true'  or episode.nikshay_id != '')
         allowed_case_types_and_users = self._allowed_case_type(episode_case) and self._allowed_user(episode_case)
-        episode_case_properties = episode_case.dynamic_case_properties()
-        return allowed_case_types_and_users and (
-            not episode_case_properties.get('nikshay_registered', 'false') == 'true' and
-            not episode_case_properties.get('nikshay_id', False) and
-            episode_pending_registration_changed(episode_case) and
-            not test_submission(episode_case)
-        )
+        if allowed_case_types_and_users:
+            episode_case_properties = episode_case.dynamic_case_properties()
+            person_case = get_person_case_from_episode(episode_case.domain, episode_case.get_id)
+
+            return (
+                not episode_case_properties.get('nikshay_registered', 'false') == 'true' and
+                not episode_case_properties.get('nikshay_id', False) and
+                episode_pending_registration_changed(episode_case) and
+                not test_submission(person_case)
+            )
+        else:
+            return False
+
+
+class NikshayHIVTestRepeater(CaseRepeater):
+    class Meta(object):
+        app_label = 'repeaters'
 
+    include_app_id_param = False
+    friendly_name = _("Forward eNikshay Patient's HIV Test to Nikshay")
 
-def test_submission(episode_case):
-    person_case = get_person_case_from_episode(episode_case.domain, episode_case.get_id)
+    @classmethod
+    def available_for_domain(cls, domain):
+        return NIKSHAY_INTEGRATION.enabled(domain)
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.nikshay.views import NikshayHIVTestRepeaterView
+        return reverse(NikshayHIVTestRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, person_case):
+        # episode.nikshay_registered is true and nikshay_id present and
+        # person.hiv_status changed OR
+        # CPTDeliverDate changes OR
+        # InitiatedDate/Art Initiated date changes
+        allowed_case_types_and_users = self._allowed_case_type(person_case) and self._allowed_user(person_case)
+        if allowed_case_types_and_users:
+            episode_case = get_open_episode_case_from_person(person_case.domain, person_case.get_id)
+            episode_case_properties = episode_case.dynamic_case_properties()
+
+            return (
+                episode_case_properties.get('nikshay_registered', 'false') == 'true' and
+                episode_case_properties.get('nikshay_id') and
```

Comment:
Expecting this to return `None`. Wont it return empty string if its empty string? so should not make a difference to add that default False. Nikshay. Also can't see a case where this should be blank string since its only added [here](https://github.com/dimagi/commcare-hq/blob/master/custom/enikshay/integrations/nikshay/repeater_generator.py#L81) and not updated again.